keps/sig-cloud-provider/20190704-instance-host-labelling.md

```diff
@@ -0,0 +1,174 @@
+
+As an application developer, I want pods to be scheduled in the same instance host so that networking between them is faster.
+I use `topology.kubernetes.io/instance-host` as topology key in [inter-pod affinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#inter-pod-affinity-and-anti-affinity-beta-feature) rules.
+
+#### Story 2
+
+As an application developer, I want pods to be spread in different instance hosts so that if the host goes down, my service doesn’t have to scale from zero.
+I use `topology.kubernetes.io/instance-host` as topology key in [even pod spreading](/keps/sig-scheduling/20190221-even-pods-spreading.md) rules.
+
+### Implementation Details/Notes/Constraints
+
+- The label will initially be named `topology.beta.kubernetes.io/instance-host` before its promoted to GA.
+
+- If the label is not explicitly used in inter-pod affinity or anti-affinity, current behavior in the scheduler remains.
+
+    There is one case worth mentioning:
+    An empty topology key for `preferredDuringSchedulingIgnoredDuringExecution` in inter-pod anti-affinity is interpreted as
+    "all topologies", so the instance host should be considered as well.
+    If the label value is empty, then the anti-affinity is still ensured by the hostname.
+    Additionally, a node [should always be considered in the same topology as itself](https://groups.google.com/d/msg/kubernetes-sig-cloud-provider/32N59IYXogY/FXIUHeYWDwAJ), regardless of the labels.
+
+### Risks and Mitigations
+
+- Risk: Some or all nodes are missing a value for the instance host label and an application developer tries to use the label as a topology key.
+
+  Mitigation: This will be documented as undefined behavior.
+  
+- Risk: If a cloud provider or hypervisor performs a live migration of a VM, the host will change.
+
+  Mitigation: The label should be updated. Evicting running pods is out-of-scope for the scheduler.
+  New pods are scheduled according to the new labels.
+
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+### Version Skew Strategy
+
+TBD
+
+## Implementation History
+
+- 2019-07-04: Initial KEP sent out for review (Summary, Motivation, Proposal, Drawbacks and Alternatives)
+
+## Drawbacks
+
+Cloud providers or other stakeholders might argue in favor of adding more well-known labels to identify other topologies (such as rack).
+Adding instance host label as a well-known label might blur the criteria for making certain topology standard versus custom or provider-specific.
+
+We argue that all cloud providers (and most on-prem clusters) have their nodes deployed as VMs running on physical hosts.
+Other topology terms don't have such clear definition.
+Thus, the instance host topology should be part of the known topologies.
+
+## Alternatives
+
+Do nothing.
```

Comment:
Actually, this sounds a bit naive but might just be ok 😀 On the other side, nothing speaks against adding this label if the use cases and impact (e.g. drift, etc.) are clearly articulated. Customers can then decide which label to use for proximity, resiliency, spreading, etc.